nncf/quantization/algo.py

```diff
@@ -929,91 +886,164 @@ def _build_commands_for_single_unified_scale_group(self,
+                                                                                  insertion_points,
+                                                                                  qconfig,
+                                                                                  range_init_minmax_values)
         return quantizer_module_id, commands
 
     def _select_final_qconfig(self, quantizer_config_list: List[QuantizerConfig]) -> QuantizerConfig:
         # Quantizer config list entries should arrive in the same order as they are listed
         # in the HW config, where they are sorted by descending order of priority
         return quantizer_config_list[0]
 
-    def _add_single_activation_quantizer(self, target_model: NNCFNetwork,
-                                         insertion_points: List[PTTargetPoint],
-                                         qconfig: QuantizerConfig,
-                                         range_init_minmax_values: Tuple[torch.Tensor, torch.Tensor] = None) -> \
-            Tuple[NonWeightQuantizerId, List[PTInsertionCommand]]:
-        """Will return one or more insertion commands - depending on whether insertion_points has one or
-        more entries. The first insertion point in the list will be associated with the actual quantizer
-        module, while the rest will still have quantization enabled, but the quantization at these points
-        will share the quantizer module with the first."""
+    def _quantize_at_points_by_single_module(self, target_model: NNCFNetwork,
+                                             insertion_points: List[PTTargetPoint],
+                                             qconfig: QuantizerConfig,
+                                             range_init_minmax_values: Tuple[torch.Tensor, torch.Tensor] = None) -> \
+            Tuple[QuantizerId, List[PTInsertionCommand]]:
+        """Will generate insertion commands for quantization at possibly multiple points
+        in the network using one and the same trainable quantizer module. The trainable
+        quantizer module will be saved either inside the weightable module which weights
+        it quantizes (for single-point weight quantization), or in a NNCFNetwork wrapper
+        module (i.e. in a storage external to the original module).
+        :param: target_model - the model to be quantized.
+        :param: insertion_points - a list of target points for quantization using one
+        quantizer module
+        :param: qconfig - the QuantizerConfig for the resulting quantizer module
+        :param: range_init_minmax_values - a pair of minimum and maximum values of input statistics
+        for initializing the quantizer's trainable parameters
+        :return: A tuple with the identifier of the new quantizer module and a list of
+        insertion commands registering this module for quantization at spots described by
+        insertion_points."""
+        #pylint:disable=too-many-branches
```

Comment:
it seems this function can be splitted into two to process activation and weight separately without significant code duplication

Reply:
If it *could* be split, I would have left the two functions. As it is, it cannot be split because it is now required to share a single quantizer module between both activation and weight quantization points in general. If you have a solution that I am not seeing, please let me know.

Reply:
why not to process weights first, return quantized module and pass it to the next function for processing activations?

Reply:
The function looks terrifying, I agree, but I do not currently see a solution that would make it more beautiful code-wise without overdesigning. Perhaps in the future, if a new use case appears, it will be more clear as to how to decompose the function's body into potentially reusable blocks.